### PR TITLE
Fix Trans component to support react 18 types

### PR DESCRIPTION
**Pull request by pedrodurek**
Closes https://github.com/i18next/react-i18next/issues/1490 and https://github.com/i18next/react-i18next/issues/1483

#### Checklist

- [x] only relevant code is changed (make a diff before you submit the PR)
- [ ] run tests `npm run test`
- [ ] tests are included

#### Checklist (for documentation change)

- [ ] only relevant documentation part is changed (make a diff before you submit the PR)
- [ ] motivation/reason is provided

### CODE REVIEW

**Comment by adrai** [comment]
included in v11.16.8

File: ts4.1/index.d.ts [hunk 1]
```diff
@@ -97,6 +99,15 @@ declare module 'i18next' {
...
 }
 
+type ObjectOrNever = TypeOptions['allowObjectInHTMLChildren'] extends true
+  ? Record<string, unknown>
+  : never;
+declare module 'react' {
+  interface HTMLAttributes {
+    children?: ReactNode | ObjectOrNever;
+  }
+}
```
Comment (cellog) [inline ts4.1/index.d.ts:109]: this override is breaking all components that use render props when upgrading to React 18. We can patch this away, but I would strongly encourage NOT overriding generic types for all components. Can't it be limited to just the `Trans` component? A sledgehammer can't be the only solution when a scalpel will do?

Comment (adrai) [reply ts4.1/index.d.ts:109]: Feel free to provide a PR with a better solution that works.

// @marcalexiei do you know about this?

Comment (pedrodurek) [reply ts4.1/index.d.ts:109]: @cellog **extending** is different from **overriding**. Share a minimal reproducible example so we can see how we can help you.

**Pull request merged**